utils/utils.go

```diff
@@ -231,6 +231,88 @@ func GeneratePasswd(username string, uid, gid uint32, homedir, rootfs, rundir st
 	return passwdFile, nil
 }
 
+// GenerateGroup generates a container specific group file,
+// iff gid is not defined in the containers /etc/group
+func GenerateGroup(gid uint32, rootfs, rundir string) (string, error) {
```

Comment:
can you refactor the shared bits between these 4 functions (Generate* and Get*)?